### PR TITLE
Adding task server to give preference to task url than tfs url.

### PR DESCRIPTION
**Pull request by kasubram**
…is important when running a release.

Fixed release extension to get rooted path.
Cleaned up variables in release extension.

### CODE REVIEW

**Comment by kasubram** [comment]
This is the same behaviour as the old agent.
Without this, getTaskDefnitionsAsync will return "api {guid} is not registered with http://server.vsrm.visualstudio.com/."
When running release the vssConnection is created with rm service URL. But for task definition it has to reach tfs.

File: src/Microsoft.VisualStudio.Services.Agent/TaskServer.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        Task<TaskDefinition> GetTaskDefinitionAsync(Guid taskId, TaskVersion taskVersion, CancellationToken token);
+    }
+
+    public sealed class TaskServer : AgentService, ITaskServer
+    {
+        private bool _hasConnection;
+
+        private VssConnection _connection;
+
+        private TaskHttpClient _taskClient;
```
Comment (TingluoHuang) [inline src/Microsoft.VisualStudio.Services.Agent/TaskServer.cs]: you don't need this.

File: src/Microsoft.VisualStudio.Services.Agent/TaskServer.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+namespace Microsoft.VisualStudio.Services.Agent
+{
+    [ServiceLocator(Default = typeof(TaskServer))]
+    public interface ITaskServer : IAgentService
+    {
+        Task ConnectAsync(VssConnection jobConnection);
+
+        // task download
+        Task<Stream> GetTaskContentZipAsync(Guid taskId, TaskVersion taskVersion, CancellationToken token);
+        Task<TaskDefinition> GetTaskDefinitionAsync(Guid taskId, TaskVersion taskVersion, CancellationToken token);
```
Comment (TingluoHuang) [inline src/Microsoft.VisualStudio.Services.Agent/TaskServer.cs]: maybe remove this? it not been used at all. GetTaskDefinitionAsync()

**Comment by TingluoHuang** [comment]
Looks good, merge after address comments. :D

File: src/Microsoft.VisualStudio.Services.Agent/TaskServer.cs (hunk 2 above)
Comment (kasubram) [inline src/Microsoft.VisualStudio.Services.Agent/TaskServer.cs]: Removed..

File: src/Microsoft.VisualStudio.Services.Agent/TaskServer.cs (hunk 1 above)
Comment (kasubram) [inline src/Microsoft.VisualStudio.Services.Agent/TaskServer.cs]: correct.. removed it. Thanks.

**Pull request merged**